pype/lib.py

```diff
@@ -708,3 +709,607 @@ def __init__(self):
+                        "repres": [
+                            <RepreEntity1>, <RepreEntity2>, ...
+                        ]
+                    }
+                },
+                ...
+            }
+        },
+        ...
+    }
+    output[asset_id]["subsets"][subset_id]["version"]["repres"]
+    ```
+    """
+
+    if not asset_entities:
+        return {}
+
+    asset_entity_by_ids = {asset["_id"]: asset for asset in asset_entities}
+
+    subsets = list(io.find({
+        "type": "subset",
+        "parent": {"$in": asset_entity_by_ids.keys()}
+    }))
+    subset_entity_by_ids = {subset["_id"]: subset for subset in subsets}
+
+    sorted_versions = list(io.find({
+        "type": "version",
+        "parent": {"$in": subset_entity_by_ids.keys()}
+    }).sort("name", -1))
+
+    subset_id_with_latest_version = []
+    last_versions_by_id = {}
+    for version in sorted_versions:
+        subset_id = version["parent"]
+        if subset_id in subset_id_with_latest_version:
+            continue
+        subset_id_with_latest_version.append(subset_id)
+        last_versions_by_id[version["_id"]] = version
+
+    repres = io.find({
+        "type": "representation",
+        "parent": {"$in": last_versions_by_id.keys()}
+    })
+
+    output = {}
+    for repre in repres:
+        version_id = repre["parent"]
+        version = last_versions_by_id[version_id]
+
+        subset_id = version["parent"]
+        subset = subset_entity_by_ids[subset_id]
+
+        asset_id = subset["parent"]
+        asset = asset_entity_by_ids[asset_id]
+
+        if asset_id not in output:
+            output[asset_id] = {
+                "asset_entity": asset,
+                "subsets": {}
+            }
+
+        if subset_id not in output[asset_id]["subsets"]:
+            output[asset_id]["subsets"][subset_id] = {
+                "subset_entity": subset,
+                "version": {
+                    "version_entity": version,
+                    "repres": []
+                }
+            }
+
+        output[asset_id]["subsets"][subset_id]["version"]["repres"].append(
+            repre
+        )
+
+    return output
+
+
+def load_containers_to_workfile():
+    """Load containers for (first) workfile.
+
+    Loads latest versions of current and linked assets to workfile by logic
+    stored in Workfile variants from presets. Variants are set by host,
+    filtered by current task name and used by families.
+
+    Each family can specify representation names and loaders for
+    representations and first available and successful loaded representation is
+    returned as container.
+
+    At the end you'll get list of loaded containers per each asset.
+
+    loaded_containers [{
+        "asset_entity": <AssetEntity1>,
+        "containers": [<Container1>, <Container2>, ...]
+    }, {
+        "asset_entity": <AssetEntity2>,
+        "containers": [<Container3>, ...]
+    }, {
+        ...
+    }]
+    """
+    io.install()
```

Comment:
What's the reason `io.install()` needs to be explicitly triggered in this function? Shouldn't it have been installed prior to calling this?